src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,112 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.exception.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    @Override
+    public Product create(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "INSERT INTO products(name, price) VALUES (?, ?)",
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setBigDecimal(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not add item to DB", e);
+        }
+        return item;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        Product product;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM products WHERE deleted=0 AND product_id=?");
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            long productId = rs.getInt("product_id");
+            String name = rs.getString("name");
+            BigDecimal price = rs.getBigDecimal("price");
+            product = new Product(productId, name, price);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get item from DB", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can not get product from DB with id - " + id, e);
```
in other words — try to add method parameter to exception message `cause in most cases thats the reason of exception